src/vsr/replica.zig

```diff
@@ -2945,28 +2945,22 @@ pub fn ReplicaType(
                 return;
             }
 
+            // Use the opposite direction from `replicate()`.
+            const ring_direction: i16 = if (prepare.message.header.op % 2 == 0) -1 else 1;
```

Comment:
Hm, I think we want the _same_ direction here? You want to jump over the faulty replica to the first healthy one, to resume the ring. If you jump to the last replica, the ring will remain unclosed? 

It's only for _next_ op that it makes sense to switch direction: 

- backups switch direction as well, so it _will_ get replicated across the entire ring
- and it will additionally inform backups about the previous prepare.

Reply:
If I am missing the reason why we do the reverse here, would be good to add that "why" to the comment!

Reply:
:facepalm: Ah you're right -- and I actually changed it back to the same direction yesterday but forgot to force-push!